Fix module names and formatting

adamdmharvey opened the pull request:
The new modules placed in v5.1(unreleased) weren't aligned to the actual names, and one had a plurality difference. Formatted v5.0 and v3.0 version modules listing with bullets for consistency. Added some headers for formality. Didn't touch proposed versions.

### Description
Fixes typos of module names listed in the v5.1 section introduced partially by ca3aee51, and enhances formatting of the rest of the CHANGELOG.

### Check List
- n/a [ ] New functionality includes testing.
- n/a [ ] All tests pass. (`$ ./build.sh`).
- n/a [ ] New functionality has been documented in the README if applicable.
- n/a [ ] New functionality has been thoroughly documented in the examples (please include helpful comments).
- Documentation only update.

soodpr commented:
@fgbulsoni - After reviewing the comments given in the Pull request #301, I have added the 5.1.0 version. But as you have mentioned that 5.0.0 is not yet released, we should merge notes of 5.1.1 in 5.0.0.

adamdmharvey commented:
Per @soodpr feedback, took a stab.  Removed v5.1 and merged the content into v5.0,, changed v5.0 to say (unreleased).  And minor, but removed the headings for the "Notes" so that after each version heading there is just always a sentence to describe the version - seemed cleaner, and matched what was listed in there for v1.0.  Hope this makes sense & just trying to help out.  😄
